### PR TITLE
Added saving viewer window status and button status to drtpref.txt file

### PR DESCRIPTION
**Pull request by sparker256**
Have added feature to save viewer window status and buttons status to drtpref.txt file in ./Resources/plugins/DataRefTool. This will allow DRT to remember where it was when X-Plane shut down.

### CODE REVIEW

File: src/viewer_window.cpp [hunk 1]
```diff
@@ -345,6 +345,20 @@ class DatarefViewerWindow {
...
 		XPSetWidgetProperty(scroll_bar, xpProperty_ScrollBarMax, 0);
 
+        FILE * pFile;
+        pFile = fopen ("./Resources/plugins/DataRefTool/drtpref.txt","r+");
+        int tmp_last_case_sensitive = last_case_sensitive;
+        int tmp_last_regex = last_regex;
+        int tmp_last_changed = last_changed;
+
+        if (pFile) {
+           fscanf(pFile,"%d %d %d %d %d %d %d", &last_left, &last_top, &last_right, &last_bottom, &tmp_last_case_sensitive, &tmp_last_regex, &tmp_last_changed);
```
Comment (leecbaker) [inline src/viewer_window.cpp]: What if the file isn't in the format we expect? We should check the fscanf return value.

File: src/viewer_window.cpp [hunk 2]
```diff
@@ -372,6 +386,13 @@ class DatarefViewerWindow {
 		last_regex = 0 != XPGetWidgetProperty(regex_toggle_button, xpProperty_ButtonState, nullptr);
 		last_changed = 0 != XPGetWidgetProperty(change_filter_button, xpProperty_ButtonState, nullptr);
 
+        FILE * pFile;
+        pFile = fopen ("./Resources/plugins/DataRefTool/drtpref.txt","w+");
+        if (pFile != NULL) {
+            fprintf(pFile,"%d %d %d %d %d %d %d\n",last_left, last_top, last_right, last_bottom, last_case_sensitive, last_regex, last_changed);
```
Comment (leecbaker) [inline src/viewer_window.cpp]: It's good practice to check the fprintf return value here.

File: src/viewer_window.cpp [hunk 3]
```diff
@@ -372,6 +386,13 @@ class DatarefViewerWindow {
 		last_regex = 0 != XPGetWidgetProperty(regex_toggle_button, xpProperty_ButtonState, nullptr);
 		last_changed = 0 != XPGetWidgetProperty(change_filter_button, xpProperty_ButtonState, nullptr);
 
+        FILE * pFile;
+        pFile = fopen ("./Resources/plugins/DataRefTool/drtpref.txt","w+");
```
Comment (leecbaker) [inline src/viewer_window.cpp]: Shouldn't we use Xplane's XPLMGetPrefsPath() to figure out the preferred directory?

Comment (sparker256) [inline src/viewer_window.cpp]: I like to keep the prefs file in the plugin folder so it does not get lost in the ./Output/preferences folder. It is a method I have used for a long time over all platforms.

**Pull request closed without merging**